sql/src/test/java/io/crate/testing/QueryTester.java

```diff
@@ -0,0 +1,339 @@
+            nodeEnvironment = new NodeEnvironment(Settings.EMPTY, env);
+            luceneReferenceResolver = new LuceneReferenceResolver(
+                mapperService::fullName,
+                idxSettings
+            );
+            IndexService indexService = indexModule.newIndexService(
+                nodeEnvironment,
+                NamedXContentRegistry.EMPTY,
+                new IndexService.ShardStoreDeleter() {
+                    @Override
+                    public void deleteShardStore(String reason, ShardLock lock, IndexSettings indexSettings) throws IOException {
+
+                    }
+
+                    @Override
+                    public void addPendingDelete(ShardId shardId, IndexSettings indexSettings) {
+
+                    }
+                },
+                new NoneCircuitBreakerService(),
+                BigArrays.NON_RECYCLING_INSTANCE,
+                threadPool,
+                scriptService,
+                client,
+                new IndicesQueryCache(Settings.EMPTY),
+                mapperRegistry,
+                new IndicesFieldDataCache(Settings.EMPTY, mock(IndexFieldDataCache.Listener.class)),
+                namedWriteableRegistry
+            );
+            indexFieldDataService = indexService.fieldData();
+            IndexWriterConfig conf = new IndexWriterConfig(new StandardAnalyzer());
+            writer = new IndexWriter(new RAMDirectory(), conf);
+            QualifiedName tableName = new QualifiedName(table.ident().name());
+            DocTableRelation docTableRelation = new DocTableRelation(table);
+            expressions = new SqlExpressions(
+                Collections.singletonMap(tableName, docTableRelation),
+                docTableRelation
+            );
+            DirectoryReader reader = DirectoryReader.open(writer, true, true);
+            queryShardContext.set(new QueryShardContext(
+                0,
+                idxSettings,
+                bitsetFilterCache,
+                indexFieldDataService::getForField,
+                Builder.this.mapperService,
+                similarityService,
+                scriptService,
+                NamedXContentRegistry.EMPTY,
+                namedWriteableRegistry,
+                client,
+                reader,
+                System::currentTimeMillis,
+                "dummyClusterAlias"
+            ));
+        }
+
+        private DocSchemaInfo findDocSchema(Schemas schemas) {
+            for (SchemaInfo schema : schemas) {
+                if (schema instanceof DocSchemaInfo) {
+                    return (DocSchemaInfo) schema;
+                }
+            }
+            throw new IllegalArgumentException("Create table statement must result in the creation of a user table");
+        }
+
+        public Builder addDoc(String column, Object value) throws IOException {
```

Comment:
I think this is a bit confusing (as "doc" is overloaded with all sorts of meanings already).

IMO `addValue` / `addColumnValue` would be better suited. 

Also, would it make sense to expose the value as vararg?